src/bluetooth/VaccineDataAccess.js

```diff
@@ -48,12 +48,18 @@ export class VaccineDataAccess {
 
   getSensors = () => this.db.objects(VACCINE_ENTITIES.SENSOR);
 
-  upsertBreaches = breaches =>
+  upsertBreaches = breaches => {
     this.db.write(() => {
       breaches.forEach(breach => {
-        this.db.update(VACCINE_ENTITIES.TEMPERATURE_BREACH, breach);
+        this.db.update(VACCINE_ENTITIES.TEMPERATURE_BREACH, {
+          ...breach,
+          id: breach.id,
+          startTimestamp: new Date(breach.startTimestamp),
+          endTimestamp: breach.endTimestamp ? new Date(breach.endTimestamp) : breach.endTimestamp,
+        });
```

Comment:
Ensuring dates are Date's when saving to realm

Reply:
So many different kinds of dates and objects 😕  eep. I think the default here is (just a little) counter-intuitive, more explicit to just set to `null` if no `endTimestamp`?